plugins/experimental/ja3_fingerprint/ja3_fingerprint.cc

```diff
@@ -0,0 +1,514 @@
+/** @ja3_fingerprint.cc
+  Plugin JA3 Fingerprint calculates JA3 signatures for incoming SSL traffic.
+  @section license License
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include <inttypes.h>
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
+#include <cerrno>
+#include <stdlib.h>
+#include <getopt.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+
+#include <algorithm>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <unordered_set>
+#include <unordered_map>
+#include <memory>
+#include <regex>
+
+#include "ts/ts.h"
+#include "ts/remap.h"
+
+// Access SSL internal if OpenSSL is less than 1.1.0
+#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#ifdef OPENSSL_NO_SSL_INTERN
+#undef OPENSSL_NO_SSL_INTERN
+#endif
+#endif
+
+#include <openssl/ssl.h>
+#include <openssl/md5.h>
+#include <openssl/opensslv.h>
+
+namespace
+{
+#define ELLIPTIC_CURVE_POINT_TYPE 0x0a
+#define ELLIPTIC_CURVE_POINT_FORMAT_TYPE 0x0b
+
+// Get 16bit big endian order and update pointer
+inline unsigned int
+n2s(const unsigned char *c)
+{
+  unsigned int s = (c[0] << 8) | (c[1]);
+  c += 2;
```

Comment:
That is pointless, as `c` is pass by value. This would be useful if the argument was declared as `const unsigned char *&c`. I'm not sure it's worth the effort, but this might be a good place to use `string_view`, passing it by reference to update via `remove_prefix`.